lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -271,6 +289,9 @@ export function usePolledAPIFetch<T = unknown>({
     timeout.current = null;
   }, [_clearTimeout]);
 
+  const refreshCount = useRef(0);
```

Comment:
What will happen if the path or other parameters change while a request is in-flight? It looks to me like `result.isLoading` will transition to `true`, the effect will bail out without resetting `refreshCount` and then when the result comes back we may continue incrementing the refresh count from the previous step?

Reply:
Hm, yeah, you are right. There's more cases in which the counter should be reset.

Reply:
I see an even worst problem here. When path or params change while we are still polling the previous request, we should reset `prevData`. Currently, we are leaking the last data fetched from previous URL/params, until the first attempt for the new path/params finishes.

Reply:
> I see an even worst problem here. When path or params change while we are still polling the previous request, we should reset `prevData`. Currently, we are leaking the last data fetched from previous URL/params, until the first attempt for the new path/params finishes.

I have refactored a bit the logic in `usePolledAPIFetch` to address this.

I have added a `refreshing` reference which is set to true right before a refresh is scheduled, and set to `false` after `result.isLoading`.

That way, when `result.isLoading` transitions to `true`, we can know if it's because a refresh was just triggered, or if it is because of an unrelated change in `path` and/or `params`. If it's the latter, we reset `prevDate` and the `refreshCount`.